### DISCUSSION TITLE
Can we use dynamic routing with getStaticPaths()?

### DISCUSSION THREAD

**Discussion by mattfranciswork0** [nextauthjs/next-auth#2705] [Help]
Situation:
I have a folder named listing, inside the folder I have [types].js

I only want movies and tvs to be valid [types].js

Such as localhost:3000/listing/movies and localhost:3000/listing/movies should be the only valid routes

Code:

```
 import React, { useState, useEffect } from "react";
        import { getProviders } from "next-auth/client";
        import Head from "next/head";
        import Header from "../../components/Header";

        import { useSession } from "next-auth/client";

        export default function Index({ providers }) {
            const [session, loading] = useSession();

            return (
                <main>
                    <Head>
                        <title>Nextjs | Next-Auth</title>
                        <link rel="icon" href="/favicon.ico" />
                    </Head>
                    <Header providers={providers} />
                </main>
            );
        }
        // This is the recommended way for Next.js 9.3 or newer
        export async function getServerSideProps(context) {
            const providers = await getProviders();
            return {
                props: { providers },
            };
        }
     //Having this would give an error of "you can not use getStaticProps or getStaticPaths with getServerSideProps. To use SSG, please remove getServerSideProps"
        export const getStaticPaths = async () => {
            return {
                paths: [
                    { params: { types: "tvs" } },
                    { params: { types: "movies" } },

                ],
                fallback: false, // fallback is set to false because we already know the slugs ahead of time
            };
        };
```

**Comment by balazsorban44** [comment]
You cannot use `getServerSideProps` and `getStaticPaths` together. See https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation

I think you meant to ask this here instead: https://github.com/vercel/next.js/discussions

**Reply by mattfranciswork0** [reply]
Thanks you for respondoing @balazsorban44 , how could we use next-auth with dynamic routing that only allows specific route urls than? Since next-auth relies on getServerSideProps?

**Reply by balazsorban44** [reply]
next-auth does not rely on getServerSideProps. It relies on a safe backend, like Next.js's API routes.

You don't need to use getServerSideProps for the sake of next-auth. I suggest you read up on the purpose of data fetching in Next.js first. Also, try to study our docs at https://next-auth.js.org to understand its usecases and ways to use it.
